clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1198,13 +1207,40 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
 
     LastParamIndex = Params.size();
     Params.push_back(NewParam);
+    BufferLocationMD.push_back(LocationID);
+  }
+
+  // Obtain an integer value stored in a template parameter of buffer_location
+  // property to pass it to buffer_location kernel attribute
+  size_t handleBufferLocationProperty(QualType FieldTy) {
+    const auto *AccTy =
+        cast<ClassTemplateSpecializationDecl>(FieldTy->getAsRecordDecl());
+
+    // TODO: when SYCL headers' part is ready - replace this 'if' with an assert
```

Comment:
This cannot be an assert, since we don't necessarily control the headers.  This should likely be a diagnostic.